src/formatDate.js

```diff
@@ -8,7 +8,42 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let dateArr = [];
+  let resultString = '';
+  let index = 0;
+
+  dateArr = date.split(fromFormat[3]);
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const fromIndex = fromFormat.indexOf(toFormat[i]);
+
+    index++;
+
+    if (toFormat[i] === 'YY' && fromFormat.includes('YYYY')) {
+      const number = dateArr[index - 1];
+
+      resultString += number.slice(-2);
+    } else if (toFormat[i] === 'YYYY' && fromFormat.includes('YY')) {
+      const year = Number(dateArr[index - 1]);
+
+      if (dateArr[index - 1] === '00') {
+        resultString += '2000.';
+        continue;
+      } else {
+        const fullYear = year < 30 ? `20${year}` : `19${year}`;
+
+        resultString += fullYear;
+      }
+    } else {
+      resultString += dateArr[fromIndex];
+    }
```

Comment:
Here you're adding a date part from `dateArr` to `resultString`. However, the separator is not added after the date part. This can lead to incorrect results if the date part is not the last part in the new format. Consider adding the separator after the date part, if it is not the last part in the new format.